internal/featuregates/featuregates.go

```diff
@@ -42,3 +48,59 @@ func (fg *FeatureGates) IsEnabled(ctx context.Context, feature string) bool {
 	}
 	return false
 }
+
+// Method to read the feature specific config parameters from the configmap
+// The feature specific config params are stored in their own configmap like this
+// data: |
+//  key1=value1
+//  key2=value2
+
+func (fg *FeatureGates) GetFeatureGateParams(ctx context.Context, feature string) map[string]string {
```

Comment:
Since this is featuregate specific it's better to be this way. Also gives us the flexibility to modify the implementation with changes to feature gate implementations. If a generic function makes sense then it should go in generic util imho.